### ISSUE TITLE
3.0.0-rc001 release

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#939]
Can be labelled **ready** when all other issues destined for this release are closed.

- [x] run code analysis in VS in *Release* mode and address violations (send a regular PR which must be merged before continuing)
- [x] change `VERSION_SUFFIX` in [AppVeyor](https://ci.appveyor.com/project/FakeItEasy/fakeiteasy/settings) to "-rc001" and initiate a build
- [x] check build
- edit draft release in [GitHub UI](https://github.com/FakeItEasy/FakeItEasy/releases):
    - [x] ensure completeness of release notes, including non-owner contributors, if any
    - [x] attach main nupkg and analyzer nupkg
    - [x] publish the release
- [x] push nupkgs to NuGet using the Deploy option in AppVeyor
- [x] tweet, mentioning contributors
- [x] post link to tweet as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] ~~if another pre-release is planned, run `rake pre_release[new_version]` to~~
    - ~~create a new draft GitHub Release~~
    - ~~create a new issue (like this one) for the next release, adding it to the next RTM milestone~~
- [x] close this issue

**Comment by blairconrad** [comment]
With the discussion on #962 winding down, I think we may soon be in a position to release this issue. Is there anything else the other owners want to get in?

Me, I'd be happy with what we have. In fact, I'd suggest we go to 3.0.0 (or -rc001, if y'all want).

**Comment by adamralph** [comment]
The only other breaking issue in the backlog is #798 but I don't think that's worth tackling for 3.0.0. I'm happy to set the wheels in motion after we've solved #962. I think it may be worth releasing an RC, since it's a major release. What do you think?

**Comment by thomaslevesque** [comment]
> The only other breaking issue in the backlog is #798 but I don't think that's worth tackling for 3.0.0.

We're deprecating self initializing fakes in 3.0.0, so we shouldn't actually remove them until a later major version.

> I think it may be worth releasing an RC, since it's a major release.

I agree. It will also be the only chance to field-test #962.

**Comment by blairconrad** [comment]
Excellent. Let's convert this to a 3.0.0-rc01 once we've solved #962!

**Comment by thomaslevesque** [comment]
Should it be `rc1` or `rc001`? (I don't think we need 3 digits, but this is the pattern we used for alphas and betas...)

**Comment by adamralph** [comment]
I think it's simpler to use the same convention for all pre-releases (that's what we do over at NServiceBus).

If only NuGet could start to support SemVer. Then we could use `3.0.0-rc.1`.

**Comment by blairconrad** [comment]
Sounds like we're ready to release, no?

**Comment by thomaslevesque** [comment]
> Sounds like we're ready to release, no?

I think we are!

**Comment by adamralph** [comment]
:shipit:

**Comment by godrose** [comment]
Does this version support the latest Castle.Core?

**Comment by thomaslevesque** [comment]
@godrose yes, 3.0.0-rc001 will support Castle.Core 4.0.0

**Comment by blairconrad** [comment]
https://twitter.com/fakeiteasyfx/status/830008518694535168

https://twitter.com/fakeiteasyfx/status/830008643735130116
